core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -344,7 +344,7 @@ private synchronized Optional<DistributedStagesScheduler> createDistributedStage
         if (queryStateMachine.isDone()) {
             return Optional.empty();
         }
-        if (attempt > 0 && retryPolicy == RetryPolicy.QUERY) {
```

Comment:
I think we rename the commit message as `Remove redundant check when creating DistributedStagesScheduler`

Reply:
updated